scipy/sparse/linalg/_dsolve/linsolve.py

```diff
@@ -269,8 +269,10 @@ def spsolve(A, b, permc_spec=None, use_umfpack=True):
             else:
                 flag = 0  # CSR format
 
+            indices = A.indices.astype(np.intc, copy=False)
```

Comment:
Should this not include `casting='safe'`?

Reply:
Good point, added.